src/bot.py

```diff
@@ -267,10 +268,10 @@ def doit(self, bot: Bot):
                     return
                 bot._curr_location = Location.NIHLATHAK_PORTAL
                 wait(0.2, 0.4)
-                self.success &= bot._char.select_by_template(["A5_RED_PORTAL", "A5_RED_PORTAL_TEXT"], expect_loading_screen=True)
+                portal_found = bot._char.select_by_template(["A5_RED_PORTAL", "A5_RED_PORTAL_TEXT"], expect_loading_screen=True)
                 time.sleep(0.5)
-                self.success &= bot._template_finder.search_and_wait(["PINDLE_0", "PINDLE_1"], threshold=0.65, time_out=20)[0]
-                if not self.success:
+                pindle_start_found = bot._template_finder.search_and_wait(["PINDLE_0", "PINDLE_1"], threshold=0.65, time_out=20)
+                if not (portal_found and pindle_start_found):
```

Comment:
why you dont use self.success here anymore? It is checked afterwards to determine if we want to end run or end game.

Reply:
I didn't see it get reused at the end, must have missed it. I only did it here because you can't do arithmetic &= on objects. Will rework according to our discord discussion.

Reply:
Now that I'm using .valid I can use the original logic.